### PR TITLE
triangle.py

### PR DESCRIPTION
**Pull request by amymhaddad**

### CODE REVIEW

File: triangle.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+"""Tests to find out triangle type: equilateral, isosceles, or scalene"""
+
+def make_set(sides):
```
Comment (amymhaddad) [inline triangle.py]: @paulghaddad When I refactored, I created a new function b/c I found myself creating a set for each function (equilateral, isosceles, scalene). This reduces the repetition. Is this the best option?

Comment (paulghaddad) [reply triangle.py]: This is probably overkill as `make_set(sides)` and `set(sides)` does the same thing.

File: triangle.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        return True
+    return False
+
+def is_isosceles(sides):
+    """Test to see if a triangle is an isosceles triangle"""
+
+    isosceles_set = make_set(sides)
+    two_equal_side_lengths = len(isosceles_set) <= 2
+
+    if sides[0] + sides[1] < sides[2]:
```
Comment (paulghaddad) [inline triangle.py]: So what is this `if` statement doing? It's similar to line 38 (hint hint).

File: triangle.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+
+def create_set(sides):
+    """Create a set for a triangle and find its length"""
+
+    return len(set(sides))
+#set_sides = lambda side: len(set(sides))
```
Comment (amymhaddad) [inline triangle.py]: @paulghaddad Can I make this first function a lambda? Is this the right way to do it?

File: triangle.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    """
+
+    sum_sides_test_1 = sides[1] + sides[2] >= sides[0]
+    sum_sides_test_2 = sides[0] + sides[2] >= sides[1]
+    sum_sides_test_3 = sides[0] + sides[1] >= sides[2]
+    sum_two_side_lengths_greater_than_third = sum_sides_test_1 or sum_sides_test_2 or sum_sides_test_3
+    
+    triangle_set = create_set(sides)
+
+    if 0 in triangle_set or sum_two_side_lengths_greater_than_third:
```
Comment (amymhaddad) [inline triangle.py]: @paulghaddad Something's not right, and I'm not sure why. I get an error when I test if 0 is in triangle_set. Why? 

There are two tests that keep breaking my code and I believe it has to do with the inequality checks in valid_triangle. These are the tests: 

def test_violation_of_triangle_inequality_not_is_isosceles(self):
   self.assertIs(is_isosceles([1, 1, 3]), False)

def test_false_if_all_sides_are_equal(self):
   self.assertIs(is_scalene([4, 4, 4]), False)

File: triangle.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    """Test if shape is a triangle"""
+
+    sum_two_sides_less_than_third = sides[1] + sides[2] <= sides[0] or \
+                                sides[0] + sides[2] <= sides[1] or sides[0] + sides[1] <= sides[2]
+
+    if 0 in sides or sum_two_sides_less_than_third:
+        return False
+    return True
+
+def create_set(sides):
```
Comment (amymhaddad) [inline triangle.py]: @paulghaddad Could this be a lambda? I tried creating one: set_sides = lambda side: len(set(sides))

Thoughts? Would I call a lambda the same way as a function?

**Pull request closed without merging**